### PR TITLE
feat: DLT-1812 language server - initial implementation

### PR DESCRIPTION
**Pull request by juliodialpad**
# Language Server - Initial implementation

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Feature

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1812

## :book: Description

- Added `packages/language-server` package which contains all the language server logic and the vscode extension code.
- Added documentation
- Component, Prop and Values completion
- CSS Variable completion
- The Extension is built with ESBuild

## :bulb: Context

We're building a VSCode extension to help developers to autocomplete Dialtone components and CSS Variables.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

- Provide hover functionality
- Improve prop suggestion
- Detect multi-line components
- Publish the extension to the Visual Studio Marketplace

## :camera: Screenshots / GIFs

- Component completion
Screenshot 2024-10-30 at 12 07 28 p m

- Property completion
Screenshot 2024-10-30 at 12 07 46 p m

- Property values completion
Screenshot 2024-10-30 at 12 07 55 p m

## :link: Sources

- <https://code.visualstudio.com/api/language-extensions/embedded-languages>
- <https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-embedded-language-service>
- <https://volarjs.dev/core-concepts/why-volar/>

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ packages:
   # documentation sites
   - 'apps/*'
   # npm publishable packages
-  - 'packages/**/*'
+  - 'packages/*'
+  - 'packages/dialtone-icons/*'
+  - 'packages/language-server/*'
```
Comment (braddialpad) [inline pnpm-workspace.yaml:7]: Wouldn't the previous `'packages/**/*'` cover this?

File: .vscode/extensions.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "stylelint.vscode-stylelint",
     "jock.svg",
     "vitest.explorer",
-    "vue.volar"
+    "vue.volar",
+    "johnsoncodehk.volarjs-labs"
```
Comment (braddialpad) [inline .vscode/extensions.json:18]: This for volar development?

**Review by braddialpad — APPROVED** [review]
Alright I really have no business reviewing this so I'm just gonna let you merge it. Definitely needs to be documented in the future.

File: pnpm-workspace.yaml (hunk 1 above)
Comment (juliodialpad) [reply pnpm-workspace.yaml:7]: That was too general, I was seeing unnamed packages that I didn't knew where they were coming from, so this way is more controlled.

File: .vscode/extensions.json (hunk 2 above)
Comment (juliodialpad) [reply .vscode/extensions.json:18]: Yes

**Pull request merged**